pkg/clean/types.go

```diff
@@ -3,3 +3,27 @@ package clean
 type cleanup interface {
 	Cleanup() error
 }
+
+type CleanupService struct {
+	cleans []cleanup
+	err    error
+}
+
+func NewCleanupService() *CleanupService {
+	return &CleanupService{}
+}
+
+func (c *CleanupService) AddCleanupService(f func() (cleanup, error)) *CleanupService {
+	var clean cleanup
+	if c.err != nil {
+		return c
+	}
+
```

Comment:
it seems that not necessary to use chain pattern, check the err in line 18 seems strange

Reply:
It's important since it would return the error and skip the following steps in advance.
An elegant pattern.